integration/spark/shared/src/main/java/io/openlineage/spark/agent/util/JdbcSparkUtils.java

```diff
@@ -39,13 +39,26 @@ public static <D extends OpenLineage.Dataset> List<D> getDatasets(
     String jdbcUrl = relation.jdbcOptions().url();
     Properties jdbcProperties = relation.jdbcOptions().asConnectionProperties();
 
+    if (meta.columnLineage().isEmpty() && meta.inTables().size() > 1) {
```

Comment:
how come empty column level lineage makes here a difference? Could you add some comment?

Reply:
I ve tried to not make any regression. It might be that the condition is not necessary in this form.

Reply:
I ve changed that a little in newest commit, the issue is with how we handle the schema when we have more than one table and we dont have column lineage. In that case we need for each column to set schema to empty StructType or parse plan (but I think thats what column lineage will at the end ll do)